src/js/angular/core/directives/queryeditor/query-editor.directive.js

```diff
@@ -574,21 +586,66 @@ function queryEditorDirective($timeout, $location, toastr, $repositories, Sparql
             window.yasr = yasr;
             yasr.afterCopy = afterCopy;
             yasr.getQueryResultsAsFormat = function (downloadFormat) {
-                // Simple cross-browser download with a form
-                const $wbDownload = $('#wb-download');
-                $wbDownload.attr('action', 'repositories/' + $repositories.getActiveRepository());
-                $('#wb-download-query').val(scope.currentQuery.query);
-                if (window.editor.getValue() !== scope.currentQuery.query) {
+                const exportFormatWithErrorHandling = scope.exportFormats.filter((format) => format.name.includes('JSON'));
+                const apiUrl = 'repositories/' + $repositories.getActiveRepository();
+                const query = scope.currentQuery.query;
+                const infer = scope.currentQuery.inference;
+                const sameAs = scope.currentQuery.sameAs;
+                const auth = localStorage.getItem('com.ontotext.graphdb.auth');
+                const accept = downloadFormat;
+
+                const fetchWithErrorHandling = exportFormatWithErrorHandling.some((format) => format.type === downloadFormat);
+                if (window.editor.getValue() !== query) {
                     toastr.warning($translate.instant('query.editor.query.results.mismatch'));
                 }
-                $('#wb-download-infer').val(scope.currentQuery.inference);
-                $('#wb-download-sameAs').val(scope.currentQuery.sameAs);
-                const auth = localStorage.getItem('com.ontotext.graphdb.auth');
-                if (auth) {
-                    $('#wb-auth-token').val(auth);
+                if (fetchWithErrorHandling) {
+                    // JSON and JSON-LD have to be fetched in memory before sending, so error handling must be introduced
+                    $http.get(apiUrl, {
```

Comment:
Can you move this in some of the rest controllers?

Reply:
Moved it to RepositoriesRestService